### PR TITLE
Enable core clr in all test projects

### PR DESCRIPTION
**Pull request by jmloeffler**
I think I was able to get `dnxcore50` supported in all of the test projects.  I'm not sure if everything is exactly identical as it was before but the tests do still pass.  In particular, I wasn't sure about the `XmlTextWriter` replacement and the way I removed Moq from `OpenIdConnectMiddlewareTests`

#454

### CODE REVIEW

File: test/Microsoft.AspNet.Authentication.Test/OpenIdConnect/OpenIdConnectMiddlewareTests.cs [hunk 1]
```diff
@@ -151,20 +150,17 @@ public class OpenIdConnectMiddlewareTests
...
-            var mockOpenIdConnectMessage = new Mock<OpenIdConnectMessage>();
-            mockOpenIdConnectMessage.Setup(m => m.CreateAuthenticationRequestUrl()).Returns(ExpectedAuthorizeRequest);
-            mockOpenIdConnectMessage.Setup(m => m.CreateLogoutRequestUrl()).Returns(ExpectedLogoutRequest);
             options.AutomaticAuthentication = true;
             options.Events = new OpenIdConnectEvents()
             {
                 OnRedirectToAuthenticationEndpoint = (context) =>
                 {
-                    context.ProtocolMessage = mockOpenIdConnectMessage.Object;
+                    context.ProtocolMessage = new OpenIdConnectMessage(ExpectedAuthorizeRequest);
```
Comment (Eilon) [inline test/Microsoft.AspNet.Authentication.Test/OpenIdConnect/OpenIdConnectMiddlewareTests.cs]: I'm not sure how that message type works, but this certainly *looks* like it's different behavior. The mock was returning these request objects only when certain methods were called.

File: test/Microsoft.AspNet.Authentication.Test/Cookies/CookieMiddlewareTests.cs [hunk 2]
```diff
@@ -1023,14 +1023,9 @@ private static void Describe(HttpResponse res, AuthenticateContext result)
             {
                 xml.Add(result.Properties.Select(extra => new XElement("extra", new XAttribute("type", extra.Key), new XAttribute("value", extra.Value))));
             }
-            using (var memory = new MemoryStream())
```
Comment (davidfowl) [inline test/Microsoft.AspNet.Authentication.Test/Cookies/CookieMiddlewareTests.cs]: Why this change?

File: test/Microsoft.AspNet.Authentication.Test/Cookies/CookieMiddlewareTests.cs [hunk 3]
```diff
@@ -1023,14 +1023,9 @@ private static void Describe(HttpResponse res, AuthenticateContext result)
...
-            {
-                using (var writer = new XmlTextWriter(memory, Encoding.UTF8))
-                {
-                    xml.WriteTo(writer);
-                }
-                res.Body.Write(memory.ToArray(), 0, memory.ToArray().Length);
-            }
+
+            var xmlBytes = Encoding.UTF8.GetBytes(xml.ToString());
```
Comment (Eilon) [inline test/Microsoft.AspNet.Authentication.Test/Cookies/CookieMiddlewareTests.cs]: This looks fine to me. It strikes me as something where either it totally works or it totally doesn't...

File: test/Microsoft.AspNet.Authentication.Test/Cookies/CookieMiddlewareTests.cs (hunk 2 above)
Comment (jmloeffler) [inline test/Microsoft.AspNet.Authentication.Test/Cookies/CookieMiddlewareTests.cs]: It did not appear that there was support for XmlTextWriter in dnxcore50.  I needed to replace it with something that was compatible.  Is there a better way?  I wasn't sure about these.  The same code is duplicated in multiple places, but that is a whole other issue.

File: test/Microsoft.AspNet.Authentication.Test/OpenIdConnect/OpenIdConnectMiddlewareTests.cs (hunk 1 above)
Comment (jmloeffler) [inline test/Microsoft.AspNet.Authentication.Test/OpenIdConnect/OpenIdConnectMiddlewareTests.cs]: It is most certainly different but seems to get the same result.  I suspect that I'm abusing some internal behavior of OpenIdConnectMessage, but without a mocking framework to directly replace the return values from those methods I was at a loss for what to do.  This seemed to make the endpoint behave the same way in the tests so I went with it.

I heard that a supported version of moq was coming, but is there an alternative in the meantime?

Comment (Eilon) [inline test/Microsoft.AspNet.Authentication.Test/OpenIdConnect/OpenIdConnectMiddlewareTests.cs]: You could create a "manual mock" by deriving from `OpenIdConnectMessage`. In that derived constructor you could take in these two values as parameters, and then override the message's virtual methods to return the instances passed in the constructor. That's effectively what Moq is doing.

And yes, a compatible version of Moq is coming, but not soon enough :smile:

**Comment by Tratcher** [comment]
@jmloeffler your git history was really messy so I recreated the changes locally and merged them. Thanks!

**Comment by jmloeffler** [comment]
Sorry about that @Tratcher.  I appreciate the extra work though.  I'm sure it's not your favorite part of the job having to re-implement somebody else's changes.  Any tips on keeping the history clean?  Once I submitted the PR I didn't see anything I could do aside from keep merging in upstream/dev and correcting things along the way.

In any case, sorry about the additional headache.

**Comment by Tratcher** [comment]
Rebase and force push is the best way to keep a PR branch current. It was pretty easy to replicate you're changes, you'd already solved all the problems.
